src/Core/Managed/Shared/Extensibility/Implementation/TaskTimer.cs

```diff
@@ -17,15 +17,22 @@ namespace Microsoft.ApplicationInsights.Extensibility.Implementation
     /// <summary>
     /// Runs a task after a certain delay and log any error.
     /// </summary>
-    public class TaskTimer : IDisposable
+    public class TaskTimer
```

Comment:
I generally agree, but I wanted to focus on Disposable right here since technically making it internal is still a breaking change.